src/vfb_query_builder/query_roller.py

```diff
@@ -259,6 +259,15 @@ def ep_stage(self):
 
     # IMAGES
 
+    def anat_cluster_dataset(self):
+        return Clause(
+            MATCH=Template("MATCH ($pvar$labels)<-[:composed_primarily_of]-(c:Cluster)" \
+                           "-[:has_source]->(ds:scRNAseq_DataSet)"),
+            WITH="%s AS cluster, %s AS dataset" 
+                 "" % (roll_min_node_info("c"), roll_min_node_info("ds")),
+            vars=["cluster, dataset"]
+        )
+
```

Comment:
Added Clause to lib to pull cluster and source dataset.  This consists of 
* a MATCH statement, with a slot for a primary variable name (pvar)
* A WITH clause, that specifies the JSON to be returned (minimal info for cluster and datasets nodes, rolled using standard function)
* A list of vars for future WITH clauses and the final RETURN clause.